flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/PrintUtils.java

```diff
@@ -174,27 +209,264 @@ public static void printAsTableauForm(
+                    }
+                    return formattedRow;
+
+                } else if (fieldType instanceof RowType && field instanceof RowData) {
+                    RowData rowData = (RowData) field;
+                    Row formattedRow = new Row(rowData.getRowKind(), rowData.getArity());
+                    for (int i = 0; i < ((RowType) fieldType).getFields().size(); i++) {
+                        LogicalType type = ((RowType) fieldType).getFields().get(i).getType();
+                        RowData.FieldGetter fieldGetter = RowData.createFieldGetter(type, i);
+                        formattedRow.setField(
+                                i,
+                                formattedTimestamp(
+                                        fieldGetter.getFieldOrNull(rowData),
+                                        type,
+                                        sessionTimeZone));
+                    }
+                    return formattedRow;
+                } else {
+                    return field;
+                }
+            case MAP:
+                LogicalType keyType = ((MapType) fieldType).getKeyType();
+                LogicalType valueType = ((MapType) fieldType).getValueType();
+                if (fieldType instanceof MapType && field instanceof Map) {
+                    Map<Object, Object> map = ((Map) field);
+                    Map<Object, Object> formattedMap = new HashMap<>(map.size());
+                    for (Object key : map.keySet()) {
+                        formattedMap.put(
+                                formattedTimestamp(key, keyType, sessionTimeZone),
+                                formattedTimestamp(map.get(key), valueType, sessionTimeZone));
+                    }
+                    return formattedMap;
+                } else if (fieldType instanceof MapType && field instanceof MapData) {
+                    MapData map = ((MapData) field);
+                    Map<Object, Object> formattedMap = new HashMap<>(map.size());
+                    Object[] keyArray =
+                            (Object[]) formattedTimestamp(map.keyArray(), keyType, sessionTimeZone);
+                    Object[] valueArray =
+                            (Object[])
+                                    formattedTimestamp(
+                                            map.valueArray(), valueType, sessionTimeZone);
+                    for (int i = 0; i < keyArray.length; i++) {
+                        formattedMap.put(keyArray[i], valueArray[i]);
+                    }
+                    return formattedMap;
+                } else {
+                    return field;
+                }
+            default:
+                return field;
+        }
+    }
+
+    /**
+     * Formats the print content of TIMESTAMP and TIMESTAMP_LTZ type data, consider the user
+     * configured time zone.
+     */
+    private static Object formatTimestampData(
```

Comment:
Would be better to call this method `formatTimestampField`. It sounds like formatting a `TimestampData` parameter.